### PR TITLE
Add Show in FileSystem option in the inspector

### PR DESCRIPTION
**Pull request by KoBeWi**
Allows to easily locate currently opened resource. Example:

https://user-images.githubusercontent.com/2223172/210617278-435f62f5-ef74-4cb5-9c89-a8718e5e92d5.mp4

The option is disabled if resource is built-in.

### CODE REVIEW

File: editor/inspector_dock.cpp [hunk 1]
```diff
@@ -86,6 +87,15 @@ void InspectorDock::_menu_option_confirm(int p_option, bool p_confirmed) {
 		case RESOURCE_EDIT_CLIPBOARD: {
 			_paste_resource();
 		} break;
+		case RESOURCE_SHOW_IN_FILESYSTEM: {
+			ObjectID current_id = EditorNode::get_singleton()->get_editor_selection_history()->get_current();
+			Object *current_obj = current_id.is_valid() ? ObjectDB::get_instance(current_id) : nullptr;
+
+			ERR_FAIL_COND(!Object::cast_to<Resource>(current_obj));
+
+			Ref<Resource> current_res = Ref<Resource>(Object::cast_to<Resource>(current_obj));
```
Comment (bruvzg) [inline editor/inspector_dock.cpp:96]: Is `id` check necessary? `get_instance` should work with null ID, so it's probably will work like this:
```suggestion
			Ref<Resource> current_res = Ref<Resource>(Object::cast_to<Resource>(ObjectDB::get_instance(current_id)));

			ERR_FAIL_COND(current_res.is_null());
```

**Review by bruvzg — APPROVED** [review]

File: editor/inspector_dock.cpp (hunk 1 above)
Comment (KoBeWi) [reply editor/inspector_dock.cpp:96]: This code is used in a few places in this file, I just copied it.
See #70923

**Comment by KoBeWi** [comment]
Rebased after #70923
I used `_get_current_resource()` to simplify the implementation.

**Review by akien-mga — APPROVED** [review]

**Comment by akien-mga** [comment]
Thanks!

**Comment by Flavelius** [comment]
This doesn't seem to work for instantiated scenes. 'Show in filesystem' is greyed out for them (4.0 beta 16)

**Comment by KoBeWi** [comment]
Not sure what do you mean. This is for resources, not nodes.

**Comment by Flavelius** [comment]
i was expecting, as an instantiated scene is backed by a packedscene which is a resource aswell, that by viewing an instance of that in the inspector i could navigate to its filesystem reference aswell. But i see that this is a slightly different case (one has to navigate into the instance/open it first to do that which then uses the feature implemented here).

**Comment by KoBeWi** [comment]
You can open the instanced scene and right click the scene tab.

**Comment by Flavelius** [comment]
I hadn't discovered this until now, interesting. that will do it.

**Pull request merged**